File: services/backend/plan_service/src/main/java/com/example/plan_service/presentation/controllers/PlanController.java
package com.example.plan_service.presentation.controllers;

import com.example.plan_service.business.interfaces.PlanServiceInterface;
import com.example.plan_service.presentation.http.MyError;
import com.example.plan_service.presentation.http.Response;
import com.example.plan_service.presentation.requests.UserPlanRequest;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.scheduling.annotation.Async;
import org.springframework.web.bind.annotation.*;

import java.util.concurrent.CompletableFuture;

@RestController
@RequestMapping("/api/planning")
public class PlanController {
    private final PlanServiceInterface planService;

    public PlanController(PlanServiceInterface planService) {
        this.planService = planService;
    }

    @Async
    @RequestMapping(value = "/plan/user/{idUser}", method = RequestMethod.POST)
    @ResponseBody
    public CompletableFuture<ResponseEntity<Object>> addUserPlan(
            @PathVariable Integer idUser,
            @RequestBody UserPlanRequest data){
        Response<Object> result = planService.createUserPlan(idUser, data);
        if (result.getSuccessfulOperation()){
            return CompletableFuture.completedFuture(new ResponseEntity<>(result, HttpStatus.valueOf(result.getCode())));
        }
        else{
            MyError tmp = new MyError(result.getCode(), result.getMessage());
            return CompletableFuture.completedFuture(new ResponseEntity<>(tmp, HttpStatus.valueOf(tmp.getCode())));
        }
    }

    @Async
    @RequestMapping(value = "/plan/user/{idUser}", method = RequestMethod.GET)
    @ResponseBody
    public CompletableFuture<ResponseEntity<Object>> getPlan(@PathVariable Integer idUser){
        Response<Object> result = planService.getPlan(idUser);
        if (result.getSuccessfulOperation()){
            return CompletableFuture.completedFuture(new ResponseEntity<>(result, HttpStatus.valueOf(result.getCode())));
        }
        else{
            MyError tmp = new MyError(result.getCode(), result.getMessage());
            return CompletableFuture.completedFuture(new ResponseEntity<>(tmp, HttpStatus.valueOf(tmp.getCode())));
        }
    }

    @Async
    @RequestMapping(value = "/plan/user/{idUser}", method = RequestMethod.DELETE)
    @ResponseBody
    public CompletableFuture<ResponseEntity<Object>> deleteUserPlan(@PathVariable Integer idUser){
        Response<Object> result = planService.deletePlan(idUser);
        if (result.getSuccessfulOperation()){
            return CompletableFuture.completedFuture(new ResponseEntity<>(null, HttpStatus.valueOf(result.getCode())));
        }
        else{
            MyError tmp = new MyError(result.getCode(), result.getMessage());
            return CompletableFuture.completedFuture(new ResponseEntity<>(tmp, HttpStatus.valueOf(tmp.getCode())));
        }
    }
}

File: services/backend/orchestrator_service/src/main/kotlin/com/example/orchestrator_service/presentation/controllers/planning/PlanController.kt
package com.example.orchestrator_service.presentation.controllers.planning

import com.example.orchestrator_service.business.interfaces.PlanServiceInterface
<<<<<<< HEAD
import com.example.orchestrator_service.business.models.plan.request.CreateUserPlanRequest
=======
import com.example.orchestrator_service.business.models.plan.UserPlanRequest
>>>>>>> 5edfeaa93e45b6511bdfec2da5e30a5c4ef73913
import com.example.orchestrator_service.presentation.http.MyError
import kotlinx.coroutines.runBlocking
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.scheduling.annotation.Async
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["/api/backend/planning"])
class PlanController {
    @Autowired
    lateinit var planService: PlanServiceInterface

    @RequestMapping("/plan/user", method = [RequestMethod.GET])
    @ResponseBody
    @Async
<<<<<<< HEAD
    fun getUserPlan(@RequestHeader(name="Authorization", required = false, defaultValue = "") token: String): ResponseEntity<Any> = runBlocking {
        val result = planService.getUserPlan(token)
=======
    fun getUserPlan(@RequestHeader(name="Authorization", required = false, defaultValue = "") token: String,
                    @PathVariable id: Int): ResponseEntity<Any> = runBlocking {
        val result = planService.getUserPlan(id, token)
>>>>>>> 5edfeaa93e45b6511bdfec2da5e30a5c4ef73913
        if (result.successfulOperation){
            ResponseEntity.status(result.code).body(result)
        } else {
            ResponseEntity.status(result.code).body(MyError(code = result.code, error = result.error, info = result.message))
        }
    }

    @RequestMapping("/plan/user", method = [RequestMethod.POST])
    @ResponseBody
    @Async
    fun createUserPlan(@RequestHeader(name="Authorization", required = false, defaultValue = "") token: String,
                       @RequestBody data: CreateUserPlanRequest
    ): ResponseEntity<Any> = runBlocking {
        val result = planService.createUserPlan(data, token)
        if (result.successfulOperation){
            ResponseEntity.status(result.code).body(result)
        } else {
            ResponseEntity.status(result.code).body(MyError(code = result.code, error = result.error, info = result.message))
        }
    }

    @RequestMapping("/plan/user", method = [RequestMethod.DELETE])
    @ResponseBody
    @Async
    fun deleteUserPlan(@RequestHeader(name="Authorization", required = false, defaultValue = "") token: String): ResponseEntity<Any> = runBlocking {
        val result = planService.deleteUserPlan(token)
        if (result.successfulOperation){
            ResponseEntity.status(result.code).body(result)
        } else {
            ResponseEntity.status(result.code).body(MyError(code = result.code, error = result.error, info = result.message))
        }
    }

    @RequestMapping("/plan", method = [RequestMethod.GET])
    @ResponseBody
    @Async
    fun getPlans(@RequestHeader(name="Authorization", required = false, defaultValue = "") token: String): ResponseEntity<Any> = runBlocking {
        val result = planService.getPlans()
        if (result.successfulOperation){
            ResponseEntity.status(result.code).body(result)
        } else {
            ResponseEntity.status(result.code).body(result.data)
        }
    }

    @RequestMapping("/plan/user/{idUser}", method = [RequestMethod.POST])
    @ResponseBody
    @Async
    fun createUserPlan(@RequestHeader(name="Authorization", required = false, defaultValue = "") token: String,
                       @PathVariable idUser: Int,
                       @RequestBody data: UserPlanRequest): ResponseEntity<Any> = runBlocking {
        val result = planService.createPlan(idUser, data)
        if (result.successfulOperation){
            ResponseEntity.status(result.code).body(result)
        } else {
            ResponseEntity.status(result.code).body(result.data)
        }
    }
}